homeassistant/components/axis.py

```diff
@@ -236,7 +238,9 @@ def setup_device(hass, config):
             if not enable_metadatastream:
                 enable_metadatastream = True
         else:
-            discovery.load_platform(hass, component, DOMAIN, config)
+            clean_config = config.copy()
+            del clean_config['hass']
+            discovery.load_platform(hass, component, DOMAIN, clean_config)
```

Comment:
Your handling is complet wrong. Look on https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/android_ip_webcam.py#L217-L218

You need not discovery the mjpeg camera direct but you need add a own config and the original config to discovery

Reply:
Sorry, I don't understand what you mean @pvizeli

Reply:
I don't understand the reason for this request in regards to this PR, since that is unchanged code from what has been accepted at a previous review.

Reply:
I want to clarify that the config in that function is not the hass_config, it is a slimmer device specific config which to my understanding is in the correct way